sklearn/preprocessing/tests/test_target.py

```diff
@@ -0,0 +1,263 @@
+import numpy as np
+
+from sklearn.base import clone
+from sklearn.base import BaseEstimator
+from sklearn.base import TransformerMixin
+
+from sklearn.utils.testing import assert_equal
+from sklearn.utils.testing import assert_raises_regex
+from sklearn.utils.testing import assert_allclose
+from sklearn.utils.testing import assert_warns_message
+from sklearn.utils.testing import assert_no_warnings
+
+from sklearn.preprocessing import FunctionTransformer
+from sklearn.preprocessing import TransformedTargetRegressor
+from sklearn.preprocessing import StandardScaler
+
+from sklearn.linear_model import LinearRegression, Lasso
+
+from sklearn import datasets
+
+friedman = datasets.make_friedman1(random_state=0)
+
+
+def test_transform_target_regressor_error():
+    X, y = friedman
+    # provide a transformer and functions at the same time
+    regr = TransformedTargetRegressor(regressor=LinearRegression(),
+                                      transformer=StandardScaler(),
+                                      func=np.exp, inverse_func=np.log)
+    assert_raises_regex(ValueError, "'transformer' and functions"
+                        " 'func'/'inverse_func' cannot both be set.",
+                        regr.fit, X, y)
+    # fit with sample_weight with a regressor which does not support it
+    sample_weight = np.ones((y.shape[0],))
+    regr = TransformedTargetRegressor(regressor=Lasso(),
+                                      transformer=StandardScaler())
+    assert_raises_regex(TypeError, "fit\(\) got an unexpected keyword argument"
+                        " 'sample_weight'", regr.fit, X, y,
+                        sample_weight=sample_weight)
+    # func is given but inverse_func is not
+    regr = TransformedTargetRegressor(func=np.exp)
+    assert_raises_regex(ValueError, "When 'func' is not None, 'inverse_func'"
+                        " cannot be None.", regr.fit, X, y)
+
+
+def test_transform_target_regressor_invertible():
+    X, y = friedman
+    regr = TransformedTargetRegressor(regressor=LinearRegression(),
+                                      func=np.sqrt, inverse_func=np.log,
+                                      check_inverse=True)
+    assert_warns_message(UserWarning, "The provided functions or transformer"
+                         " are not strictly inverse of each other.",
+                         regr.fit, X, y)
+    regr = TransformedTargetRegressor(regressor=LinearRegression(),
+                                      func=np.sqrt, inverse_func=np.log,
+                                      check_inverse=False)
+    # the transformer/functions are not checked to be invertible the fitting
```

Comment:
I'd drop this comment, but would make it clearer by replacing the previous statement with `regr.set_params(check_inverse=False)`